src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ChainedPartitionedRateLimiter.cs

```diff
@@ -23,21 +23,36 @@ public ChainedPartitionedRateLimiter(PartitionedRateLimiter<TResource>[] limiter
             _limiters = limiters;
         }
 
-        public override int GetAvailablePermits(TResource resource)
+        public override RateLimiterStatistics? GetStatistics(TResource resource)
         {
             ThrowIfDisposed();
-            int lowestPermitCount = int.MaxValue;
+            long lowestAvailablePermits = long.MaxValue;
+            long lowestQueuedCount = long.MaxValue;
+            long totalFailedLeases = 0;
+            long totalSuccessfulLeases = 0;
             foreach (PartitionedRateLimiter<TResource> limiter in _limiters)
             {
-                int permitCount = limiter.GetAvailablePermits(resource);
-
-                if (permitCount < lowestPermitCount)
+                if (limiter.GetStatistics(resource) is { } statistics)
                 {
-                    lowestPermitCount = permitCount;
+                    if (statistics.CurrentAvailablePermits < lowestAvailablePermits)
+                    {
+                        lowestAvailablePermits = statistics.CurrentAvailablePermits;
+                    }
+                    if (statistics.CurrentQueuedCount < lowestQueuedCount)
+                    {
+                        lowestQueuedCount = statistics.CurrentQueuedCount;
```

Comment:
I can see why it makes sense to choose the lowest available permits since we'd start rejecting requests as soon as we cross that threshold, but wouldn't a sum of currently queued requests make more sense for the aggregate `CurrentQueuedCount` for the chain?